packages/wrangler/src/versions/secrets/bulk.ts

```diff
@@ -0,0 +1,128 @@
+	return yargs
+		.positional("json", {
+			describe: `The JSON file of key-value pairs to upload, in form {"key": value, ...}`,
+			type: "string",
+		})
+		.option("name", {
+			describe: "Name of the Worker",
+			type: "string",
+			requiresArg: true,
+		})
+		.option("message", {
+			describe: "Description of this deployment (optional)",
+			type: "string",
+			requiresArg: true,
+		})
+		.option("tag", {
+			describe: "A tag for this version (optional)",
+			type: "string",
+			requiresArg: true,
+		});
+}
+
+export async function versionsSecretPutBulkHandler(
+	args: StrictYargsOptionsToInterface<typeof versionsSecretsPutBulkOptions>
+) {
+	await printWranglerBanner();
+	const config = readConfig(args.config, args, false, true);
+
+	const scriptName = getLegacyScriptName(args, config);
+	if (!scriptName) {
+		const error = new UserError(
+			"Required Worker name missing. Please specify the Worker name in wrangler.toml, or pass it as an argument with `--name <worker-name>`"
+		);
+		logger.error(error.message);
+		throw error;
+	}
+
+	const accountId = await requireAuth(config);
+
+	logger.log(
+		`🌀 Creating the secrets for the Worker "${scriptName}" ${args.env ? `(${args.env})` : ""}`
+	);
+
+	let content: Record<string, string>;
+	if (args.json) {
+		const jsonFilePath = path.resolve(args.json);
+		try {
+			content = parseJSON<Record<string, string>>(
+				readFileSync(jsonFilePath),
+				jsonFilePath
+			);
+		} catch (e) {
+			return logger.error(`🚨 Please provide a JSON file or valid JSON pipe`);
+		}
+		validateJSONFileSecrets(content, args.json);
+	} else {
+		try {
+			const rl = readline.createInterface({ input: process.stdin });
+			let pipedInput = "";
+			for await (const line of rl) {
+				pipedInput += line;
+			}
+			content = parseJSON<Record<string, string>>(pipedInput);
+		} catch {
+			return logger.error(`🚨 Please provide a JSON file or valid JSON pipe`);
+		}
+	}
+
+	if (!content) {
+		return logger.error(`🚨 No content found in JSON file or piped input.`);
+	}
+
+	const secrets = Object.entries(content).map(([key, value]) => ({
+		name: key,
+		value,
+	}));
+
+	// Grab the latest version
+	const versions = (
+		await fetchResult<{ items: WorkerVersion[] }>(
+			`/accounts/${accountId}/workers/scripts/${scriptName}/versions`
+		)
+	).items;
+	if (versions.length === 0) {
+		throw new UserError(
+			"There are currently no uploaded versions of this Worker - please upload a version before uploading a secret."
+		);
+	}
+	const latestVersion = versions[0];
+
+	const newVersion = await copyWorkerVersionWithNewSecrets({
+		accountId,
+		scriptName,
+		versionId: latestVersion.id,
+		secrets,
+		versionMessage: args.message,
+		versionTag: args.tag,
+		sendMetrics: config.send_metrics,
+	});
+
+	for (const secret of secrets) {
+		logger.log(`✨ Successfully created secret for key: ${secret.name}`);
+	}
+	logger.log(
+		"\nFinished processing secrets JSON file:" +
```

Comment:
is there any point, in all these log/warn/error messages in this file, at which it would be worth mentioning the JSON file path? Just a nit, not a must, if it's even valuable to begin with

Reply:
Well the user is providing the path so probably not too much value in showing it again. Not sure where we'd add it either tbh

Reply:
fair!